### PR TITLE
Enable hosts to compile strong-named assemblies on all platforms by supp...

### LINKED ISSUE #567: Add ability to provide a CryptoKeyFile as a byte[]
Today, to provide a strong name, it has to come from a file on disk:

```C#
var options = context.CSharpCompilation.Options.WithStrongNameProvider(strongNameProvider)
                                                           .WithCryptoKeyFile(keyFile);
```

Instead, I'd like to pass a fixed byte[].

### PR DESCRIPTION
**Pull request by tmat**
...lying public key via compilation options and also enable OSS signing with such key

Fixes #567

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/VBResources.resx [hunk 1]
```diff
@@ -178,7 +178,10 @@
     <value>option '{0}' can be followed only by '+' or '-'</value>
   </data>
   <data name="ERR_InvalidSwitchValue" xml:space="preserve">
-    <value>the value '{0}' is invalid for option '{1}'</value>
+    <value>the value '{1}' is invalid for option '{0}'</value>
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/VBResources.resx]: Any particular reason you swapped this around? This is just going to confuse the translators.

File: src/Compilers/VisualBasic/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.vb [hunk 2]
```diff
@@ -301,19 +301,94 @@ End Class
...
+    End Sub
+
+    <Fact>
     Public Sub PubKeyFileBogusOptions()
         Dim tmp = Temp.CreateFile().WriteAllBytes(New Byte() {1, 2, 3, 4})
         Dim other As VisualBasicCompilation = CreateCompilationWithMscorlib(
 <compilation>
-    <file>
-        <![CDATA[
+                                                  <file>
```
Comment (gafter) [inline src/Compilers/VisualBasic/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.vb]: Lots of gratuitous indentation in this source file.

File: src/Compilers/VisualBasic/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.vb [hunk 3]
```diff
@@ -1639,10 +1714,10 @@ End Class]]>
     Public Sub Bug1095618()
         Dim source As XElement = _
 <compilation name="a">
-    <file name="a.vb"><![CDATA[
+                                                                                                                                                                                                                                                                                  <file name="a.vb"><![CDATA[
 <Assembly: System.Runtime.CompilerServices.InternalsVisibleTo("System.Runtime.Serialization, PublicKey = 10000000000000000400000000000000")>
     ]]></file>
-</compilation>
+                                                                                                                                                                                                                                                                              </compilation>
```
Comment (gafter) [inline src/Compilers/VisualBasic/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.vb]: Don't know why, but indentation gets worse and worse in this file.

**Comment by gafter** [comment]
Other than the strange indentation issue, looks good to me.

File: src/Compilers/VisualBasic/Portable/VBResources.resx (hunk 1 above)
Comment (tmat) [inline src/Compilers/VisualBasic/Portable/VBResources.resx]: Yes, to match C# order -- I need to create the diagnostic in shared code and C# had a different order.

**Comment by tmat** [comment]
@gafter I fixed the indentation. Not sure why the VB formatter decided to insert it.

**Comment by jaredpar** [comment]
Should we abstract all of the Crypto fields into a separate type?  It seems fragile to have properties where the doc (paraphrased) says:

> if this is non-null then this other property must be null

Feel like this will be much easier to manage if it's a separate type which encapsulates the various states the data can be in.

**Comment by tmat** [comment]
@davidfowl Sounds like a bug in JSON serializer.

**Comment by tmat** [comment]
@jaredpar I don't think that's really necessary. It's trivial to validate the requirement as report diagnostics, which is what we are doing.

**Comment by tmat** [comment]
@davidfowl I guess that it can be serialized if the CryptoPublicKey is specified?

**Pull request merged**